src/mt/header.rs

```diff
@@ -0,0 +1,408 @@
+///   (normal ring alert rules apply)
+/// * Update SSD Location: Update SSD location with given lat/lon values
+/// * High Priority Message: Place the associated MT payload in front of queue
+/// * Assign MTMSN: Use the value in the Unique ID field as the MTMSN
+///
+/// # Notes
+///
+/// * The bit 3 was not defined at this point, skipping from 2nd to 4th.
+///   Therefore, all flags on would be 0b0000_0000_0011_1011.
+struct DispositionFlags {
+    flush_queue: bool,
+    send_ring_alert: bool,
+    update_location: bool,
+    high_priority: bool,
+    assign_mtmsn: bool,
+}
+
+impl DispositionFlags {
+    /// Decode a u16 into a DispositionFlags
+    ///
+    /// Each flag is encoded by a bit in a specific position, which is on
+    /// (true) or off (false). Parsing the that sequence of bits, assuming
+    /// a big endian unsigned integer, results in the following values when
+    /// activated:
+    ///
+    /// * Flush MT Queue: 1
+    /// * Send Ring Alert - Mo MTM: 2
+    /// * Update SSD Location: 8
+    /// * High Priority Message: 16
+    /// * Assign MTMSN: 32
+    ///
+    /// # Notes:
+    ///
+    /// - All non used bits are ignored. It might be useful to consider a more
+    ///   strict approach, where this would fail if a non-expected bit is
+    ///   activated.
+    fn decode(code: u16) -> Self {
+        let flush_queue = match code & 0b0000_0000_0000_0001 {
+            0 => false,
+            _ => true,
+        };
+        let send_ring_alert = match code & 0b0000_0000_0000_0010 {
+            0 => false,
+            _ => true,
+        };
+        let update_location = match code & 0b0000_0000_0000_1000 {
+            0 => false,
+            _ => true,
+        };
+        let high_priority = match code & 0b0000_0000_0001_0000 {
+            0 => false,
+            _ => true,
+        };
+        let assign_mtmsn = match code & 0b0000_0000_0010_0000 {
+            0 => false,
+            _ => true,
+        };
+
+        DispositionFlags {
+            flush_queue,
+            send_ring_alert,
+            update_location,
+            high_priority,
+            assign_mtmsn,
+        }
+    }
+
+    /// Parse a DispositionFlags from a Read trait
+    fn read_from<R: std::io::Read>(mut read: R) -> Result<Self, Error> {
+        let code = read.read_u16::<BigEndian>()?;
+        Ok(DispositionFlags::decode(code))
+    }
+
+    /// Encode a DispositionFlags into an u16
+    fn encode(&self) -> u16 {
+        (u16::from(self.assign_mtmsn) << 5)
+            + (u16::from(self.high_priority) << 4)
+            + (u16::from(self.update_location) << 3)
+            + (u16::from(self.send_ring_alert) << 1)
+            + u16::from(self.flush_queue)
+    }
+
+    /// Save a DispositionFlags using a Write trait
+    fn write<W: std::io::Write>(&self, wtr: &mut W) -> Result<usize, Error> {
+        wtr.write_u16::<BigEndian>(self.encode())?;
+        Ok(2)
+    }
+}
+
+#[cfg(test)]
+mod test_disposition_flags {
```

Comment:
Can you please consolidate all tests to a single `tests` module? It's not critical, but fits closer to the standard Rust module layout.

Reply:
If you have a strong opinion about it, I'm fine with consolidating, but I think it would turn into a huge single test module. Keeping it in chunks by subjects makes it more clear to debug while defining a context for new tests. Keep in mind that `mod.rs` will be completely refactored before I finish this `mt` overall module.

I used to do it like that, a single test module, but nowadays, I tend to use groups of unit tests close to the object being tested and integration tests outside, where I might combine multiple components. For instance, the [GSW-rs](https://github.com/castelao/GSW-rs/blob/main/src/volume.rs).

Just let me know what you prefer.

Reply:
If `mod.rs` will be refactored, then its fine to keep as-is for now, because I imagine the module organization will sort itself out. If this is a work-in-progress PR, can you mark it as draft? Thanks!

Reply:
I think I'm fine for now with respect to the header. I never know when I'll have another spare time, so I would rather split the work into a few objective PRs. For instance, this one was focused on the header only. I added the minimum in the `mod.rs` just to make some sense, but the `mod.rs` is a work in progress.

If you're OK with that, let's focus on the header here. The following PRs will implement the other IEIs and the required changes in the `mod.rs`.